src/main/scala/is/hail/methods/IBD.scala

```diff
@@ -151,29 +201,47 @@ object IBD {
 
   final val chunkSize = 1024
 
-  def computeIBDMatrix(vds: VariantDataset, computeMaf: Option[(Variant, Annotation) => Double], bounded: Boolean): RDD[((Int, Int), ExtendedIBDInfo)] = {
-    val unnormalizedIbse = vds.rdd.map { case (v, (va, gs)) => ibsForGenotypes(gs, computeMaf.map(f => f(v, va))) }
-      .fold(IBSExpectations.empty)(_ join _)
-
-    val ibse = unnormalizedIbse.normalized
+  def computeIBDMatrix[RPK, RK, T >: Null](vds: VariantSampleMatrix[RPK, RK, T],
+    computeMaf: Option[(RegionValue) => Double],
+    min: Option[Double],
+    max: Option[Double],
+    sampleIds: IndexedSeq[String],
+    bounded: Boolean): RDD[RegionValue] = {
 
     val nSamples = vds.nSamples
 
-    val chunkedGenotypeMatrix = vds.rdd
-      .map { case (v, (va, gs)) => gs.map(g =>
-        if (Genotype.isCalled(g))
-          IBSFFI.gtToCRep(Genotype.unboxedGT(g))
-        else
-          IBSFFI.missingGTCRep).toArray[Byte]
+    val rowType = vds.rowType
+    val unnormalizedIbse = vds.rdd2.mapPartitions { it =>
+      val view = HardCallView(rowType)
+      it.map { rv =>
+        view.setRegion(rv)
+        ibsForGenotypes(view, computeMaf.map(f => f(rv)))
+      }
+    }.fold(IBSExpectations.empty)(_ join _)
+
+    val ibse = unnormalizedIbse.normalized
+
+    val chunkedGenotypeMatrix = vds.rdd2.mapPartitions { it =>
+      val view = HardCallView(rowType)
+      it.map { rv =>
+        view.setRegion(rv)
+        Array.tabulate[Byte](view.getLength) { i =>
```

Comment:
@cseed I think you said this in person, but just to be sure, it's okay to be making arrays here to pass to `IBSFFI` (for now)?